app/presenters/application_presenter.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+class ApplicationPresenter < SimpleDelegator
+  include ActionView::Helpers::FormTagHelper
+  include ActionView::Helpers::UrlHelper
+  include Rails.application.routes.url_helpers
+
+  attr_reader :action, :controller
+
+  # Master list of fields that could be displayed on any page.
+  ALL_FIELDS = %i[].freeze
+
+  # Not displayed on the show page
+  HIDDEN_FIELDS = %i[].freeze
+
+  # Displayed on the edit page but not editable
+  READ_ONLY_FIELDS = %i[].freeze
+
+  # Only editable when creating a new object
+  READ_ONLY_AFTER_PERSISTED_FIELDS = %i[].freeze
+
+  # Default character size of editable fields
+  FIELD_SIZE = 60
+
+  def self.attribute_scope
+    @attribute_scope ||= "activerecord.attributes." + name.sub(%r{Presenter$}, "").underscore
+  end
+
+  def self.value_scope
+    @value_scope ||= name.sub(%r{Presenter$}, "").underscore + ".values"
+  end
+
+  def self.field_label(field)
+    I18n.t(field, scope: attribute_scope)
+  end
+
+  def initialize(obj, controller: nil, action: :show)
+    super obj
+    @controller = controller
```

Comment:
Is the reason we need the controller here so that we have the URL helpers available?

Reply:
:+1: